### PR TITLE
Update JDBCConnectionImpl.java

### PR DESCRIPTION
**Pull request by fiorenzino**
Can you add "public" attribute to this class, so that the class becomes extensible?

### CODE REVIEW

**Comment by vietj** [comment]
hi, can we know the use case ?

I'm not against it (actually I hate libraries that prevents me to do stuff like this :-) ), but I'm curious to know what it's good for

**Comment by fiorenzino** [comment]
Hi,
i wrote an email about my idea (simplify the use of sql/jdbc libraries):
https://groups.google.com/forum/#!topic/vertx/YkdDWRUTf2U

I need to change two classes:
class JDBCConnectionImpl implements SQLConnection ( =>public)
public final class JDBCStatementHelper (=> non final)

I would like to extend the official libraries in a separate project.

**Comment by pmlopes** [comment]
If this happens (and I am not saying it won't) it has 2 consequences:

1. We break the contract between private API and public API
2. We open a precedent to break the contract forcing us to not allow internal changes at it would break external application code

The issue here is more of an extensibility IMHO. The current factory pattern in the same interface as the public facing API doesn't allow extension by object inheritance, only extension by wrapping.

**Comment by fiorenzino** [comment]
Hi Paulo, I understand the issues and i will try in different way. My intent was write less code, not breaking anything.

**Comment by pmlopes** [comment]
The breaking issue is not on your side, the issue is say that for example, we overhaul the internal code, re-arrange methods, rename all the things in the impl package, this would be hidden from the public API but would break your code.

Also we need to be careful here, also because if say, we add module info data for JDK9 (jigsaw) all impl packages will be really invisible to the end user so you won't be able to use these packages at all.

**Comment by pmlopes** [comment]
You can also try to identify the code that you want to reuse and discuss how to promote it from the implementation side to the public side of the API

**Comment by manikantag** [comment]
I too faced this problem. 

I want to fix **JDBCCallable.convertOutputs()** to address https://github.com/vert-x3/vertx-jdbc-client/pull/45#issuecomment-425852193, but as I can't simply override that method as my custom class will not be used by **JDBCConnectionImpl**, which is instantiated in **JDBCClientImpl.getConnection(Context, Handler<AsyncResult<SQLConnection>>)**.

And **JDBCClientImpl.getConnection(Context, Handler<AsyncResult<SQLConnection>>)** is private and can't be overridden in custom class. I tried to copy that entire method into my custom**JDBCClientImplFixed extends JDBCClientImpl**  implementation, but I can't get `ds`, `exec`, and other private members of that class. So, I had to copy entire `JDBCClientImpl` as `JDBCClientImplFixed` instead of just extending from it.

So, in conclusion, if we can keep members with at least `protected` scope, unless that member is implementation specific, would make extending a lot lot easier.

**Comment by vietj** [comment]
that sounds fair but we can't support this, i;e it is subject to changes and refactorings.
